hr_expense_advance_clearing/models/hr_expense.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2019 Kitti Upariphutthiphong <kittiu@ecosoft.co.th>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class HrExpense(models.Model):
+    _inherit = 'hr.expense'
+
+    advance = fields.Boolean(
+        string='Employee Advance',
+        default=False,
+    )
+
+    @api.multi
+    @api.constrains('advance')
+    def _check_advance(self):
+        for expense in self:
+            if expense.advance:
+                emp_advance = self.env.ref('hr_expense_advance_clearing.'
+                                           'product_emp_advance')
+                if not emp_advance.property_account_expense_id:
+                    raise ValidationError(
+                        _('Employee advance product has no payable account'))
+                if expense.product_id != emp_advance:
+                    raise ValidationError(
+                        _('Employee advance, selected product is not valid'))
+                if expense.tax_ids:
+                    raise ValidationError(
+                        _('Employee advance, all taxes must be removed'))
+                if expense.payment_mode != 'own_account':
+                    raise ValidationError(
+                        _('Employee advance, paid by must be employee'))
+        return True
+
+    @api.onchange('advance')
+    def onchange_advance(self):
+        self.tax_ids = False
+        self.payment_mode = 'own_account'
+        if self.advance:
+            self.product_id = self.env.ref(
+                'hr_expense_advance_clearing.product_emp_advance')
+        else:
+            self.product_id = False
+
+    @api.multi
+    def _get_account_move_line_values(self):
+        move_line_values_by_expense = super()._get_account_move_line_values()
+        # Only when do the clearing, change cr payable to cr advance
+        for sheet in self.mapped('sheet_id').filtered('advance_sheet_id'):
```

Comment:
```suggestion
        sheets = self.mapped('sheet_id').filtered('advance_sheet_id')
        if sheets.filtered(lambda sheet: sheet..advance_sheet_residual <= 0.0):
            raise ValidationError(_('Advance %s has no amount to clear') %
                                               sheet.advance_sheet_residual.name)
        for sheet in sheets:
```

This, because if you have 10 sheets, and in the number 9 the sheet do not have `advance_sheet_residual `, the raise revert the other 8 processes, then with this you save the time for that 8 processes if the 9th is incorrect